### DISCUSSION TITLE
Compatibility with Alpine Linux

### DISCUSSION THREAD

**Discussion by HirCoir** [cockpit-project/cockpit#17709] [Q&A] [answered]
Alpine Linux is a very light and powerful distro, it is directly entering the server world and installed it only occupies 30MB of RAM. It would be great if they made a port to work with Alpine Linux!

**Comment by jelly** [comment]
Does Alpine have systemd, as much of the functionality in Cockpit depends on systemd.

**Comment by HirCoir** [comment]
> Does Alpine have systemd, as much of the functionality in Cockpit depends on systemd.

Officially it has openrc, it is similar to systemd. For example:
rc-service enable x – Enables service x so that it starts automatically along with the system.
service x start – Starts service x and stops service x on reboot.
Alpine Linux is 80% similar to Gentoo ES4.0, check this out: https://wiki.alpinelinux.org/wiki/Webmin

**Comment by martinpitt** [comment, marked as answer]
I'm afraid openrc is not "similar" to systemd, it is hardly comparable at all (other than "can start system services at boot somehow". We *will not* support cockpit's web server running on non-systemd systems, but you can connect over SSH with the "remote host" functionality from a cockpit-ws installation, or with [Cockpit Client](https://github.com/flathub/org.cockpit_project.CockpitClient). Functionality will be much reduced, but depending on what you want to do it might still do something useful.

**Comment by bademux** [comment]
@martinpitt  @HirCoir 
for some time systemd package is available for alpine https://pkgs.postmarketos.org/packages?name=systemd
Postmarketos is alpine-based distro for mobile phones, so systemd isn't official. But it at least tested in action.

**Reply by jelly** [reply]
@bademux and as far as I know alpine will also include systemd (optionally) but this is all very much being worked on as far as I know.

**Reply by bademux** [reply]
nice to hear. It's always nice to have options.
It will be nice to have small VM image to test "out of the box" Cockpit experience. The distribution will be also great for homelabs.

**Reply by jelly** [reply]
For that purpose, install the Fedora server edition it has cockpit installed by default.

**Reply by bademux** [reply]
It's not small by any meanings :)
But came that way and used Ubuntu 24.04. Installation is, buggy; solvable, but still.
After that I found that deb is not "native" platform.

And here the Idea to have small distro with preinstalled Copilot and Alpine shines for that purpose.
